osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -38,6 +38,19 @@ public abstract class DrawableHitObject : SkinReloadableDrawable
         private readonly Lazy<List<DrawableHitObject>> nestedHitObjects = new Lazy<List<DrawableHitObject>>();
         public IReadOnlyList<DrawableHitObject> NestedHitObjects => nestedHitObjects.IsValueCreated ? nestedHitObjects.Value : (IReadOnlyList<DrawableHitObject>)Array.Empty<DrawableHitObject>();
 
+        /// <summary>
+        /// Whether this object should handle any user input events.
+        /// </summary>
+        public bool HandleUserInput { get; set; } = true;
+
+        public override bool HandlePositionalInput => HandleUserInput;
+
+        public override bool HandleNonPositionalInput => HandleUserInput;
```

Comment:
I think these are unnecessary and serve as an anti-optimisation: https://github.com/ppy/osu-framework/wiki/Handling-input-events#controlling-whether-input-is-to-be-handled

Reply:
ah yeah, wasn't sure about those. free to remove.